OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -635,24 +635,51 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
-				if (engineVersion < 20170507)
+				if (engineVersion < 20170528)
 				{
 					if (node.Key == "Offset" && parent.Key.StartsWith("WithHarvestOverlay", StringComparison.Ordinal))
 						RenameNodeKey(node, "LocalOffset");
 
-					if (node.Key == "LocalOffset")
+					var gridType = Exts.Lazy(() => modData.Manifest.Get<MapGrid>().Type);
```

Comment:
The Lazy here doesn't do anything since you force the value to be evaluated in the next line.